modules/nextflow/src/main/groovy/nextflow/config/ConfigValidator.groovy

```diff
@@ -155,6 +157,32 @@ class ConfigValidator {
             && !pluginScopes.children().containsKey(name)
     }
 
+    /**
+     * Determine whether a config option is a map option or a
+     * property thereof.
+     *
+     * @param names
+     */
+    private boolean isMapOption(List<String> names) {
+        return isMapOption0(SchemaNode.ROOT, names)
+            || isMapOption0(pluginScopes, names)
+    }
+
+    private static boolean isMapOption0(SchemaNode.Scope scope, List<String> names) {
```

Comment:
It is difficult to understand what this method is doing without some context. Is `names` providing the path to get the property and it is navigating throught the schema nodes starting from scope? Maybe a comment will help to better understand the method.